js/ui/internetSearch.js

```diff
@@ -54,22 +90,31 @@ const InternetSearchProvider = new Lang.Class({
     },
 
     getResultMetas: function(results, callback) {
-        let metas = results.map(function(resultId) {
+        let metas = results.map(Lang.bind(this, function(resultId) {
             let name;
             if (resultId.startsWith('uri:')) {
                 let uri = resultId.slice('uri:'.length);
                 name = _("Open \"%s\" in browser").format(uri);
             } else if (resultId.startsWith('search:')) {
                 let query = resultId.slice('search:'.length);
-                name = _("Search the web for \"%s\"").format(query);
+                let engineName = this._getEngineName();
+
+                if (engineName) {
+                    /* Translators: the first %s is the search engine name, and the second
+                     * is the search string. For instance, 'Search Google for "hello"'.
+                     */
+                    name = _("Search %s for \"%s\"").format(engineName, query);
+                } else {
+                    name = _("Search the web for \"%s\"").format(query);
```

Comment:
Another possible way to do this is use the first format string with engineName = _("the web"), which reduces the total number of words to translate.  Also, I think we should use "the internet" rather than "the web" for consistency with how we describe this in the help center:
> The global search bar makes search easy! You can look for apps, documents, photos and more, right from your desktop. If you have an internet conection, you can even look for things that are on the internet, like you would on google, with the global search tool.